apps/hosts/views.py

```diff
@@ -162,7 +167,7 @@ def challenge_host_detail(request, challenge_host_team_pk, pk):
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
     try:
-        challenge_host = ChallengeHost.objects.get(pk=pk)
+        challenge_host = get_challenge_host_model(pk)
     except ChallengeHost.DoesNotExist:
         response_data = {"error": "ChallengeHost does not exist"}
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
```

Comment:
@Kajol-Kumari You can replace the whole `try except` block with `challenge_host = get_challenge_host_model(pk)`.
See here: https://github.com/Cloud-CV/EvalAI/blob/1b1318d458c9dff4d2a0d936015e57215989ca21/apps/base/utils.py#L61-L64, the `get_model_object` already throws a `NotFound` (404). 
You need to change the code from `406` to `404` in tests as well.

Reply:
@nikochiko when i replaced that try case with similar to this, I got "NotFound" undefined and still getting the same error in test_views , though i had changed the code. Btw the lines you are referring here is the definition of get_model_object and as far as i could understand it's output is being stored in "get_challenge_host_model" created by me in views.py.PLease correct me if i am missing something.

Reply:
@Kajol-Kumari No, sorry if I wasn't clear. I didn't want you to use that code here.
What I meant was that, instead of the whole `try except` block:
```python
    try:
        challenge_host = get_challenge_host_model(pk)
    except ChallengeHost.DoesNotExist:
        response_data = {"error": "ChallengeHost does not exist"}
        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
```
you can just do:
```python
challenge_host = get_challenge_host_model(pk)
```
I mentioned the code to show that when the function `get_model_object` is called, it itself gives a 404 error (by raising `rest_framework.NotFound`), and the except block is not reached.

Reply:
My mistake, i understood something else. Thanks a lot for clarifying it @nikochiko